packages/backend/src/services/SlackService/SlackService.ts

```diff
@@ -0,0 +1,513 @@
+        ) {
+            console.debug(
+                `URL to unfurl ${linkUrl} does not belong to this siteUrl ${this.lightdashConfig.siteUrl}, ignoring.`,
+            );
+            return {
+                isValid: false,
+                url: linkUrl,
+            };
+        }
+
+        const shareUrl = new RegExp(`/share/${nanoid}`);
+        const url = linkUrl.match(shareUrl)
+            ? await this.getSharedUrl(linkUrl)
+            : linkUrl;
+
+        const dashboardUrl = new RegExp(`/projects/${uuid}/dashboards/${uuid}`);
+        const chartUrl = new RegExp(`/projects/${uuid}/saved/${uuid}`);
+        const exploreUrl = new RegExp(`/projects/${uuid}/tables/`);
+
+        if (url.match(dashboardUrl) !== null) {
+            return {
+                isValid: true,
+                lightdashPage: LightdashPage.DASHBOARD,
+                url: linkUrl,
+            };
+        }
+        if (url.match(chartUrl) !== null) {
+            return {
+                isValid: true,
+                lightdashPage: LightdashPage.CHART,
+                url: linkUrl,
+            };
+        }
+        if (url.match(exploreUrl) !== null) {
+            return {
+                isValid: true,
+                lightdashPage: LightdashPage.EXPLORE,
+                url: linkUrl,
+            };
+        }
+
+        console.debug(`URL to unfurl ${url} is not valid`);
+        return {
+            isValid: false,
+            url,
+        };
+    }
+
+    async unfurlPage(
+        url: string,
+        lightdashPage: LightdashPage,
+        imageUrl: string,
+    ) {
+        switch (lightdashPage) {
+            case LightdashPage.DASHBOARD:
+                return this.unfurlDashboard(url, imageUrl);
+            case LightdashPage.CHART:
+                return this.unfurlChart(url, imageUrl);
+            case LightdashPage.EXPLORE:
+                return unfurlExplore(url, imageUrl);
+            default:
+                return assertUnreachable(
+                    lightdashPage,
+                    `No lightdash page Slack unfurl implemented`,
+                );
+        }
+    }
+
+    async unfurl(event: any, client: any, context: any): Promise<void> {
+        event.links.map(async (l: any) => {
+            const { isValid, lightdashPage, url } = await this.parseUrl(l.url);
+
+            if (!isValid || lightdashPage === undefined || url === undefined) {
+                return;
+            }
+            try {
+                const userUuid = await getUserUuid(context);
+                const cookie = await getUserCookie(userUuid);
+                analytics.track({
+                    event: 'share_slack.unfurl',
+                    userId: event.user,
+                    properties: {
+                        pageType: lightdashPage,
+                    },
+                });
+
+                const imageId = `slack-image-${context.teamId}-${event.unfurl_id}`;
```

Comment:
Should the id include something from the `link`? In case the same event has multiple links they are going to end up with the same imageId.